File: shared/core/models/ConnectionState.kt
package com.neotun.core.models

import kotlinx.serialization.Serializable

@Serializable
enum class ConnectionState {
    DISCONNECTED,
    CONNECTING,
    CONNECTED,
    DISCONNECTING,
    ERROR
}

@Serializable
data class ConnectionStatus(
    val state: ConnectionState,
    val profile: VpnProfile? = null,
    val connectedAt: Long? = null,
    val bytesReceived: Long = 0,
    val bytesSent: Long = 0,
    val errorMessage: String? = null
)

@Serializable
data class ConnectionStats(
    val uptime: Long,
    val bytesReceived: Long,
    val bytesSent: Long,
    val speed: ConnectionSpeed
)

@Serializable
data class ConnectionSpeed(
    val downloadSpeed: Long, // bytes per second
    val uploadSpeed: Long    // bytes per second
)

File: windows/NeoTUN.Core/Models/ConnectionState.cs
using System.Text.Json.Serialization;

namespace NeoTUN.Core.Models;

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum ConnectionState
{
    Disconnected,
    Connecting,
    Connected,
    Disconnecting,
    Error
}

public record ConnectionStatus(
    ConnectionState State,
    VpnProfile? Profile = null,
    long? ConnectedAt = null,
    long BytesReceived = 0,
    long BytesSent = 0,
    string? ErrorMessage = null
);

public record ConnectionStats(
    long Uptime,
    long BytesReceived,
    long BytesSent,
    ConnectionSpeed Speed
);

public record ConnectionSpeed(
    long DownloadSpeed, // bytes per second
    long UploadSpeed    // bytes per second
);